### PR TITLE
(DOCSP-28056) Preimages in 6.0 change streams

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-28056) Preimages in 6.0 change streams

### Staged Changes

- [Database Triggers > Disable Collection-Level Preimages](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/preimages/triggers/database-triggers/#disable-collection-level-preimages)
- [Document Preimages](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/preimages/mongodb/preimages/)

### CODE REVIEW

File: source/mongodb/preimages.txt [hunk 1]
```diff
@@ -86,11 +91,13 @@ Enabling preimages for a collection applies to all collections. This
...
 
+.. _disable-collection-preimages:
+
 Disable Preimages for a Collection
 ----------------------------------
 
-You can disable collection-level preimages in the App Services UI. To disable 
-preimages for a collection:
+You can disable collection-level preimages in the App Services UI. To
+disable preimages for a collection:
```
Comment (mongodben) [inline source/mongodb/preimages.txt:100]:
```suggestion
You can disable collection-level preimages in the App Services UI.
To disable preimages for a collection:
```

super nit

File: source/triggers/database-triggers.txt [hunk 2]
```diff
@@ -591,22 +594,28 @@ multiple executions of a Trigger Function occur in event order.
...
 
-You can disable document preimages on a per-trigger basis to
-exclude the preimage from change events. Regardless of your
-trigger-level settings, a collection's oplog entries will
-continue to include preimage data unless you explicitly disable
-preimages for the collection.
+.. note::
 
-:ref:`Learn how to view and disable collection-level preimages. <preimages>`
+   To optimize individual trigger executions, you can exclude the preimage
```
Comment (mongodben) [inline source/triggers/database-triggers.txt:614]:
```suggestion
   To optimize individual trigger executions, exclude the preimage
```

File: source/triggers/database-triggers.txt [hunk 3]
```diff
@@ -591,22 +594,28 @@ multiple executions of a Trigger Function occur in event order.
...
-with no additonal overhead.
+Document preimages require your cluster to record additional data about
+each operation on a collection. The additional storage space and compute
+overhead may degrade trigger performance depending on your cluster
+configuration.
+
+Once enabled, your cluster continues to store preimages for a collection
+unless you explicitly disable them at the collection level. This means
+that your cluster will record preimages even if you delete all triggers
+on the collection.
```
Comment (mongodben) [inline source/triggers/database-triggers.txt:606]: i don't understanding this paragraph. could you clarify?

File: source/triggers/database-triggers.txt [hunk 4]
```diff
@@ -591,22 +594,28 @@ multiple executions of a Trigger Function occur in event order.
...
-continue to include preimage data unless you explicitly disable
-preimages for the collection.
+.. note::
 
-:ref:`Learn how to view and disable collection-level preimages. <preimages>`
+   To optimize individual trigger executions, you can exclude the preimage
+   from change events on a per-trigger basis. Regardless of your
+   trigger-level settings, a collection's oplog entries will continue to
+   include preimage data unless you explicitly disable preimages for the
+   collection.
```
Comment (mongodben) [inline source/triggers/database-triggers.txt:618]: i also struggle to understand this paragraph. could you clarify?

**Review by mongodben — COMMENTED** [review]
back to you!

File: source/triggers/database-triggers.txt (hunk 4 above)
Comment (nlarew) [reply source/triggers/database-triggers.txt:618]: Removed this note

File: source/triggers/database-triggers.txt (hunk 3 above)
Comment (nlarew) [reply source/triggers/database-triggers.txt:606]: Updated

File: source/triggers/database-triggers.txt [hunk 5]
```diff
@@ -591,22 +594,22 @@ multiple executions of a Trigger Function occur in event order.
...
-continue to include preimage data unless you explicitly disable
-preimages for the collection.
 
-:ref:`Learn how to view and disable collection-level preimages. <preimages>`
+Document preimages require your cluster to record additional data about
+each operation on a collection. The additional storage space and compute
+overhead may degrade trigger performance depending on your cluster
+configuration.
+
+Once enabled for any trigger on a collection, your cluster stores
```
Comment (mongodben) [inline source/triggers/database-triggers.txt:603]:
```suggestion
Once you enable preimages for any trigger on a collection, your cluster stores
```

Comment (mongodben) [reply source/triggers/database-triggers.txt:603]: preimages 2x in the sentence which is a bit meh but i think lil clearer. take it or leave it 🤷

File: source/triggers/database-triggers.txt [hunk 6]
```diff
@@ -591,22 +594,22 @@ multiple executions of a Trigger Function occur in event order.
...
-trigger-level settings, a collection's oplog entries will
-continue to include preimage data unless you explicitly disable
-preimages for the collection.
 
-:ref:`Learn how to view and disable collection-level preimages. <preimages>`
+Document preimages require your cluster to record additional data about
+each operation on a collection. The additional storage space and compute
+overhead may degrade trigger performance depending on your cluster
+configuration.
+
```
Comment (mongodben) [inline source/triggers/database-triggers.txt:602]: think it'd be beneficial to add a sentence or two stating that there are 2 separate preimage settings: 

1. collection level pre-images
2. trigger level pre-images

File: source/triggers/database-triggers.txt [hunk 7]
```diff
@@ -591,22 +594,22 @@ multiple executions of a Trigger Function occur in event order.
...
+Once enabled for any trigger on a collection, your cluster stores
+preimages for every operation on the collection. The collection-level
+setting is different than the preimages setting for individual triggers.
+This means that the collection records preimages even if you later
+delete or exclude preimages from all triggers on the collection.
```
Comment (mongodben) [inline source/triggers/database-triggers.txt:607]: taking the above comment https://github.com/mongodb/docs-app-services/pull/375/files#r1123699911 into account, the copy in this paragraph could prob be refined a bit, like deleting the sentence "The collection-level
setting is different than the preimages setting for individual triggers."

**Review by mongodben — COMMENTED** [review]
thanks for clarifying the questions. all clear to me. just 1 more set of suggestions

**Review by mongodben — APPROVED** [review]
beautiful. lgtm

**Pull request merged**